From b094c63ef001019ce4da0d3677df3e2c526a6f97 Mon Sep 17 00:00:00 2001
From: Chirag Shah <chirag@nvidia.com>
Date: Fri, 26 Feb 2021 08:31:07 -0800
Subject: [PATCH] tools: Mark reload failure when no form cli fails

if no form of the cli fails to execute, mark frr-reload
as failure so return code can be nonzero.
The similar approach is done for non no-form (add case) of the cli.

Ticket:CM-33345
Reviewed By:CCR-11287
Testing Done:

Signed-off-by: Chirag Shah <chirag@nvidia.com>

(cherry picked from commit f26070fc07a279d3a616e8f2d5d05069625f6de5)
---
 tools/frr-reload.py | 1 +
 1 file changed, 1 insertion(+)

diff --git a/tools/frr-reload.py b/tools/frr-reload.py
index a3aed7719..da240e919 100755
--- a/tools/frr-reload.py
+++ b/tools/frr-reload.py
@@ -1575,6 +1575,7 @@ if __name__ == '__main__':

                             if len(last_arg) <= 2:
                                 log.error('"%s" we failed to remove this command', ' -- '.join(original_cmd))
+                                reload_ok = False
                                 break

                             new_last_arg = last_arg[0:-1]
--
2.20.1
